Supertext.Base.Tests/Extensions/AttributeExtensionsTests.cs

```diff
@@ -175,6 +184,36 @@ public void GetAttributeOfType_With_Specifier_On_Enum_Throws_Expected_Exception(
             Assert.Fail("The expected exception was not thrown.");
         }
 
+        [TestMethod]
+        public void GetAttributesOfType_On_Enum_Returns_Expected_Values()
+        {
+            // Arrange
+            const MultipleTestEnum multipleTestEnum = MultipleTestEnum.TestValue1;
+
+            // Act
+            var vals = multipleTestEnum.GetAttributesOfType<TestEnumAttribute>();
+            
+            // Assert
+            vals.Should().NotBeNull();
+            vals.Count().Should().Be(2);
+            vals.SingleOrDefault(val => val.StringValue == "test-value-1").Should().NotBeNull();
```

Comment:
Only `.Single()` should work as well, shouldn't it?

Reply:
Yes, it should work for this test under these conditions, but what if there was a bug in the method under test: `Single` would cause an exception whereas `SingleOrDefault` would cause the test to fail.